core/scratch_block_comment.js

```diff
@@ -39,18 +39,30 @@ goog.require('goog.userAgent');
 /**
  * Class for a comment.
  * @param {!Blockly.Block} block The block associated with this comment.
+ * @param {string=} id Optional uid for comment; a new one will be generated if
+ *     not provided.
  * @param {number=} x Initial x position for comment, in workspace coordinates.
  * @param {number=} y Initial y position for comment, in workspace coordinates.
  * @param {boolean=} minimized Whether or not this comment is minimized
  *     (only the top bar displays), defaults to false.
  * @extends {Blockly.Comment}
  * @constructor
  */
-Blockly.ScratchBlockComment = function(block, x, y, minimized) {
+Blockly.ScratchBlockComment = function(block, id, x, y, minimized) {
   Blockly.ScratchBlockComment.superClass_.constructor.call(this, block);
   this.x_ = x;
   this.y_ = y;
   this.isMinimized_ = minimized || false;
+
+  this.workspace = block.workspace;
+  this.id = goog.isString(id) ? id : Blockly.utils.genUid();
```

Comment:
The id is essentially unused.  There's no way to look up the comment by ID--you're always looking it up by block ID instead.

You should insert the comment into the workspace's commentDB.  That also ensures that the workspace takes the position/size of the comment into account when getting metrics.

Reply:
Oh cool, I didn't know that re: "That also ensures that the workspace takes the position/size of the comment into account when getting metrics."

That's awesome.